subsys/bluetooth/host/conn.c

```diff
@@ -699,6 +699,29 @@ static void conn_cleanup(struct bt_conn *conn)
 	k_work_reschedule(&conn->deferred_work, K_NO_WAIT);
 }
 
+static void conn_destroy(struct bt_conn *conn, void *data)
+{
+	k_work_cancel_delayable(&conn->deferred_work);
+
+	sys_slist_init(&conn->tx_pending);
+	conn->pending_no_cb = 0;
+
+	conn_cleanup(conn);
+
+	bt_conn_set_state(conn, BT_CONN_DISCONNECTED);
+
+	/* bt_conn_unref is called at the end of bt_conn_foreach,
+	 * we set to 1 here, so it will be 0 at the end of
+	 * bt_conn_foreach.
+	 */
+	atomic_set(&conn->ref, 1);
```

Comment:
This looks very suspicious and likely wrong. The reference count should only ever be manipulated using the `bt_conn_unref()` and `bt_conn_ref()` functions, and those functions should be the only place that change the count directly using `atomic_*()`.

Reply:
Dealing with the tx_pending callbacks meant this was no longer needed, so have removed